File: KoKoslan/src/kokoslan/java/eval/KoKoEvaluator.java
/**
 @author loriacarlos@gmail.com
 
*/
package kokoslan.java.eval;

import kokoslan.java.ast.*;

public class KoKoEvaluator {

	private KoKoAst head;
	protected KoKoList args;

	public KoKoEvaluator(KoKoAst head, KoKoList args){
		this.head = head;
		this.args = args;
	}

	public KoKoValue evaluate(KoKoContext ctx){
		String name = ((KoKoId)this.head).getValue();
        if(name.equals("print"))
            printArguments(ctx);
		return null;
	}

	public void printArguments(KoKoContext ctx){
        this.args
            .stream()
            .forEach( a -> System.out.println(a.eval(ctx)));
    }

	/*
		public KoKoValue evaluate(KoKoProgram tree, KoKoContext ctx){
			return null;
		}
	*/
	
}

File: KoKoslan/src/kokoslan/kotlin/eval/KoKoEvaluator.kt
/*
    Andrey Arguedas Espinoza
    Daniela Armas Sanchez
    Michael Chen Wang
    Kimberly Olivas Delgado
*/

package kokoslan.kotlin.eval

import kokoslan.kotlin.ast.*

data class KoKoEvaluator(private var head: KoKoAst, protected var args: KoKoList) {


    fun evaluate(tree: KoKoProgram, ctx: KoKoContext): KoKoValue? {
        return null
    }


}